public/app/features/explore/Logs/Logs.tsx

```diff
@@ -427,6 +464,105 @@ class UnthemedLogs extends PureComponent<Props, State> {
 
   scrollToTopLogs = () => this.topLogsRef.current?.scrollIntoView();
 
+  prepareTableFrame = memoizeOne((frame: DataFrame): DataFrame => {
+    const { timeZone, splitOpen, range } = this.props;
+
+    const { timeIndex } = getTimeField(frame);
+    const sortedFrame = sortDataFrame(frame, timeIndex, this.state.logsSortOrder === LogsSortOrder.Descending);
+
+    const [frameWithOverrides] = applyFieldOverrides({
+      data: [sortedFrame],
+      timeZone,
+      theme: config.theme2,
+      replaceVariables: (v: string) => v,
+      fieldConfig: {
+        defaults: {
+          custom: {},
+        },
+        overrides: [],
+      },
+    });
+    // `getLinks` and `applyFieldOverrides` are taken from TableContainer.tsx
+    for (const field of frameWithOverrides.fields) {
+      field.getLinks = (config: ValueLinkConfig) => {
+        return getFieldLinksForExplore({
+          field,
+          rowIndex: config.valueRowIndex!,
+          splitOpenFn: splitOpen,
+          range: range,
+          dataFrame: sortedFrame!,
+        });
+      };
+      field.config = {
+        custom: {
+          filterable: true,
+          inspect: true,
+        },
+      };
+    }
+
+    return frameWithOverrides;
+  });
+
+  prepareTableData = async () => {
+    const { logsFrames } = this.props;
+    if (!logsFrames || !logsFrames.length) {
+      this.setState({ tableFrame: undefined });
+      return;
+    }
+    // TODO: This does not work with multiple logs queries for now, as we currently only support one logs frame.
+    let dataFrame = logsFrames[0];
+
+    const { timeIndex } = getTimeField(dataFrame);
```

Comment:
i recommend parsing the frame with https://github.com/grafana/grafana/blob/820e289073395f5aef2af07b23d390180ac55e89/public/app/features/logs/logsFrame.ts#L75

(the `getTimeField` here does not work with dataplane-logs-frames)


`logsFrame` should return the timefield as `FieldWithIndex`, so you can just take it's `.index` attribute.

Reply:
Good remark, changed in 0c099051f7a81ceb08012e8b37494577d603f269.